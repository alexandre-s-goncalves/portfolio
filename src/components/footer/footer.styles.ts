import styled from 'styled-components';
import {colors} from 'resources/colors';
import {margin} from 'resources/margins';
import {Text} from 'components/ui/text/Text';
import {Icon} from 'components/icon';
import HeartSVG from 'assets/icons/iHeart.svg';

interface ThemeProps {
  $themeDark?: boolean;
}

export const FooterContainer = styled.footer<ThemeProps>`
  display: flex;
  flex-direction: column;
  width: 100%;
  background-color: ${({$themeDark}) => ($themeDark ? colors.background05 : colors.background01)};
  padding: ${margin.large}px 120px;
  border-top: 1px solid ${({$themeDark}) => ($themeDark ? colors.darkerGray : colors.lighterGray)};
  box-sizing: border-box;
`;

export const ContentWrapper = styled.div<ThemeProps>`
  display: flex;
  gap: 150px;
  width: 100%;
  padding-bottom: ${margin.large}px;
  border-bottom: 1px solid
    ${({$themeDark}) => ($themeDark ? colors.darkerGray : colors.lighterGray)};
`;

export const InfoSection = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  max-width: 500px;
  gap: ${margin.small}px;
`;

export const BrandTitle = styled(Text).attrs<ThemeProps>(({$themeDark}) => ({
  type: 'h2',
  size: 18,
  lineHeight: 26,
  color: $themeDark ? colors.neutralXLight : colors.darkContent,
}))<ThemeProps>`
  display: flex;
  align-items: center;
  gap: ${margin.small}px;
`;

export const BioText = styled(Text).attrs<ThemeProps>(({$themeDark}) => ({
  size: 16,
  lineHeight: 24,
  color: $themeDark ? colors.grayLight : colors.gray,
}))<ThemeProps>``;

export const ContactSection = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  gap: ${margin.xxsmall}px;
`;

export const ContactTitle = styled(Text).attrs<ThemeProps>(({$themeDark}) => ({
  size: 16,
  lineHeight: 24,
  weight: 'semi-bold',
  color: $themeDark ? colors.neutralXLight : colors.darkContent,
}))`
  margin-bottom: ${margin.xxsmall}px;
`;

export const ContactText = styled(Text).attrs<ThemeProps>(({$themeDark}) => ({
  size: 16,
  lineHeight: 24,
  color: $themeDark ? colors.grayLight : colors.gray,
}))<ThemeProps>``;

export const SocialLinks = styled.div`
  display: flex;
  align-items: center;
  gap: ${margin.medium}px;
`;

export const SocialLink = styled.a`
  display: flex;
  align-items: center;
  justify-content: center;
  transition: transform 0.2s ease;

  &:hover {
    transform: scale(1.1);
  }
`;

export const LogoIcon = styled(Icon).attrs({
  width: 32,
  height: 32,
  fill: colors.black,
})``;

export const HeartIcon = styled(Icon).attrs({
  icon: HeartSVG,
  width: 24,
  height: 24,
  color: colors.red,
})``;

export const GithubIcon = styled(Icon).attrs<ThemeProps>(({$themeDark}) => ({
  width: 24,
  height: 24,
  color: $themeDark ? colors.grayLight : colors.black,
}))<ThemeProps>``;

export const LinkedinIcon = styled(Icon).attrs({
  width: 24,
  height: 24,
})``;

export const EmailIcon = styled(Icon).attrs<ThemeProps>(({$themeDark}) => ({
  width: 32,
  height: 32,
  color: $themeDark ? colors.grayLight : colors.black,
}))``;

export const CopyrightContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: ${margin.medium}px 0;
`;

export const CopyrightText = styled(Text).attrs<ThemeProps>(({$themeDark}) => ({
  size: 14,
  lineHeight: 20,
  weight: 'semi-bold',
  color: $themeDark ? colors.grayLight : colors.gray,
}))`
  display: flex;
  align-items: center;
  gap: ${margin.xxsmall}px;
`;
