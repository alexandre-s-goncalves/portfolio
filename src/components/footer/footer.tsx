import React from 'react';
import {useTranslation} from 'react-i18next';
import {useTheme} from 'context/ThemeContext';
import {namespaces} from 'utils/i18n/i18n.constants';
import GithubSVG from 'assets/icons/iGithub.svg';
import LinkedinSVG from 'assets/icons/iLinkedin.svg';
import EmailSVG from 'assets/icons/iEmail.svg';
import LogoDarkSVG from 'assets/image/logo-dark.svg';
import LogoClearSVG from 'assets/image/logo-clear.svg';
import * as S from './footer.styles';

export const Footer = () => {
  const {t} = useTranslation(namespaces.pages.footer);
  const {themeDark} = useTheme();

  return (
    <S.FooterContainer $themeDark={themeDark}>
      <S.ContentWrapper $themeDark={themeDark}>
        <S.InfoSection>
          <S.BrandTitle $themeDark={themeDark}>
            <S.LogoIcon
              data-testid={themeDark ? 'logo-clear' : 'logo-dark'}
              icon={themeDark ? LogoClearSVG : LogoDarkSVG}
            />
            {t('name')}
          </S.BrandTitle>
          <S.BioText $themeDark={themeDark}>{t('bio')}</S.BioText>
          <S.SocialLinks>
            <S.SocialLink
              href={t('github')}
              target="_blank"
              rel="noopener noreferrer"
              aria-label={t('github')}>
              <S.GithubIcon $themeDark={themeDark} icon={GithubSVG} />
            </S.SocialLink>
            <S.SocialLink
              href={t('linkedin')}
              target="_blank"
              rel="noopener noreferrer"
              aria-label={t('linkedin')}>
              <S.LinkedinIcon icon={LinkedinSVG} />
            </S.SocialLink>
            <S.SocialLink href={`mailto:${t('email')}`} aria-label="Send email">
              <S.EmailIcon $themeDark={themeDark} icon={EmailSVG} />
            </S.SocialLink>
          </S.SocialLinks>
        </S.InfoSection>
        <S.ContactSection>
          <S.ContactTitle $themeDark={themeDark}>{t('contact')}</S.ContactTitle>
          <S.ContactText $themeDark={themeDark}>{t('email')}</S.ContactText>
          <S.ContactText $themeDark={themeDark}>{t('phone')}</S.ContactText>
          <S.ContactText $themeDark={themeDark}>{t('location')}</S.ContactText>
        </S.ContactSection>
      </S.ContentWrapper>
      <S.CopyrightContainer>
        <S.CopyrightText $themeDark={themeDark}>
          Â© 2025 {t('name')} {t('rights')}
        </S.CopyrightText>
        <S.CopyrightText $themeDark={themeDark}>
          {t('built_with')} <S.HeartIcon data-testid="heart-icon" /> React
        </S.CopyrightText>
      </S.CopyrightContainer>
    </S.FooterContainer>
  );
};
