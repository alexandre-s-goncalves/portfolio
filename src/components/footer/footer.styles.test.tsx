import React from 'react';
import {render} from '@testing-library/react';
import {BrowserRouter} from 'react-router-dom';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from './footer.styles';

describe('footer.styles', () => {
  describe('FooterContainer', () => {
    test('should match the snapshot', () => {
      const {container} = render(<S.FooterContainer $themeDark={false} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should render as footer element', () => {
      const {container} = render(<S.FooterContainer $themeDark={false} />);
      const element = container.firstChild as HTMLElement;
      expect(element.nodeName).toBe('FOOTER');
    });

    test('should apply light theme background and border color', () => {
      const {getByTestId} = render(<S.FooterContainer data-testid="footer" $themeDark={false} />);
      const footer = getByTestId('footer');
      expect(footer).toHaveStyle('background-color: #ffffff');
      expect(footer).toHaveStyle('border-top: 1px solid #e5e5e5');
    });

    test('should apply dark theme background and border color', () => {
      const {getByTestId} = render(<S.FooterContainer data-testid="footer" $themeDark={true} />);
      const footer = getByTestId('footer');
      expect(footer).toHaveStyle('background-color: #0A0A0A');
      expect(footer).toHaveStyle('border-top: 1px solid #262626');
    });

    test('should have full width', () => {
      const {getByTestId} = render(<S.FooterContainer data-testid="footer" $themeDark={false} />);
      const footer = getByTestId('footer');
      expect(footer).toHaveStyle('width: 100%');
    });

    test('should have flex column layout', () => {
      const {getByTestId} = render(<S.FooterContainer data-testid="footer" $themeDark={false} />);
      const footer = getByTestId('footer');
      expect(footer).toHaveStyle('display: flex');
      expect(footer).toHaveStyle('flex-direction: column');
    });

    test('should have correct padding', () => {
      const {getByTestId} = render(<S.FooterContainer data-testid="footer" $themeDark={false} />);
      const footer = getByTestId('footer');
      expect(footer).toHaveStyle('padding: 32px 120px');
    });
  });

  describe('ContentWrapper', () => {
    test('should match the snapshot', () => {
      const {container} = render(<S.ContentWrapper $themeDark={false} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should render as div element', () => {
      const {container} = render(<S.ContentWrapper $themeDark={false} />);
      const element = container.firstChild as HTMLElement;
      expect(element.nodeName).toBe('DIV');
    });

    test('should have flex layout with gap', () => {
      const {getByTestId} = render(<S.ContentWrapper data-testid="wrapper" $themeDark={false} />);
      const wrapper = getByTestId('wrapper');
      expect(wrapper).toHaveStyle('display: flex');
      expect(wrapper).toHaveStyle('gap: 150px');
    });

    test('should apply light theme border color', () => {
      const {getByTestId} = render(<S.ContentWrapper data-testid="wrapper" $themeDark={false} />);
      const wrapper = getByTestId('wrapper');
      expect(wrapper).toHaveStyle('border-bottom: 1px solid #e5e5e5');
    });

    test('should apply dark theme border color', () => {
      const {getByTestId} = render(<S.ContentWrapper data-testid="wrapper" $themeDark={true} />);
      const wrapper = getByTestId('wrapper');
      expect(wrapper).toHaveStyle('border-bottom: 1px solid #262626');
    });

    test('should have full width', () => {
      const {getByTestId} = render(<S.ContentWrapper data-testid="wrapper" $themeDark={false} />);
      const wrapper = getByTestId('wrapper');
      expect(wrapper).toHaveStyle('width: 100%');
    });

    test('should have correct padding bottom', () => {
      const {getByTestId} = render(<S.ContentWrapper data-testid="wrapper" $themeDark={false} />);
      const wrapper = getByTestId('wrapper');
      expect(wrapper).toHaveStyle('padding-bottom: 32px');
    });
  });

  describe('InfoSection', () => {
    test('should match the snapshot', () => {
      const {container} = render(<S.InfoSection />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should render as div element', () => {
      const {container} = render(<S.InfoSection />);
      const element = container.firstChild as HTMLElement;
      expect(element.nodeName).toBe('DIV');
    });

    test('should have flex column layout', () => {
      const {getByTestId} = render(<S.InfoSection data-testid="info" />);
      const section = getByTestId('info');
      expect(section).toHaveStyle('display: flex');
      expect(section).toHaveStyle('flex-direction: column');
    });

    test('should have correct gap', () => {
      const {getByTestId} = render(<S.InfoSection data-testid="info" />);
      const section = getByTestId('info');
      expect(section).toHaveStyle('gap: 16px');
    });

    test('should have max-width constraint', () => {
      const {getByTestId} = render(<S.InfoSection data-testid="info" />);
      const section = getByTestId('info');
      expect(section).toHaveStyle('max-width: 500px');
    });

    test('should have flex: 1', () => {
      const {getByTestId} = render(<S.InfoSection data-testid="info" />);
      const section = getByTestId('info');
      expect(section).toHaveStyle('flex: 1');
    });
  });

  describe('BrandTitle', () => {
    test('should match the snapshot with light theme', () => {
      const {container} = render(<S.BrandTitle $themeDark={false}>Brand</S.BrandTitle>);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should match the snapshot with dark theme', () => {
      const {container} = render(<S.BrandTitle $themeDark={true}>Brand</S.BrandTitle>);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should apply light theme color', () => {
      const {getByText} = render(<S.BrandTitle $themeDark={false}>Brand</S.BrandTitle>);
      const title = getByText('Brand');
      expect(title).toHaveStyle('color: #0a0a0a');
    });

    test('should apply dark theme color', () => {
      const {getByText} = render(<S.BrandTitle $themeDark={true}>Brand</S.BrandTitle>);
      const title = getByText('Brand');
      expect(title).toHaveStyle('color: #F0F2F1');
    });

    test('should have flex layout with gap', () => {
      const {getByText} = render(<S.BrandTitle $themeDark={false}>Brand</S.BrandTitle>);
      const title = getByText('Brand');
      expect(title).toHaveStyle('display: flex');
      expect(title).toHaveStyle('align-items: center');
      expect(title).toHaveStyle('gap: 16px');
    });
  });

  describe('BioText', () => {
    test('should match the snapshot with light theme', () => {
      const {container} = render(<S.BioText $themeDark={false}>Bio text</S.BioText>);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should match the snapshot with dark theme', () => {
      const {container} = render(<S.BioText $themeDark={true}>Bio text</S.BioText>);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should apply light theme color', () => {
      const {getByText} = render(<S.BioText $themeDark={false}>Bio text</S.BioText>);
      const text = getByText('Bio text');
      expect(text).toHaveStyle('color: #737373');
    });

    test('should apply dark theme color', () => {
      const {getByText} = render(<S.BioText $themeDark={true}>Bio text</S.BioText>);
      const text = getByText('Bio text');
      expect(text).toHaveStyle('color: #a3a3a3');
    });
  });

  describe('ContactSection', () => {
    test('should match the snapshot', () => {
      const {container} = render(<S.ContactSection />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should render as div element', () => {
      const {container} = render(<S.ContactSection />);
      const element = container.firstChild as HTMLElement;
      expect(element.nodeName).toBe('DIV');
    });

    test('should have flex column layout', () => {
      const {getByTestId} = render(<S.ContactSection data-testid="contact" />);
      const section = getByTestId('contact');
      expect(section).toHaveStyle('display: flex');
      expect(section).toHaveStyle('flex-direction: column');
    });

    test('should have correct gap', () => {
      const {getByTestId} = render(<S.ContactSection data-testid="contact" />);
      const section = getByTestId('contact');
      expect(section).toHaveStyle('gap: 4px');
    });

    test('should have flex: 1', () => {
      const {getByTestId} = render(<S.ContactSection data-testid="contact" />);
      const section = getByTestId('contact');
      expect(section).toHaveStyle('flex: 1');
    });
  });

  describe('ContactTitle', () => {
    test('should match the snapshot with light theme', () => {
      const {container} = render(<S.ContactTitle $themeDark={false}>Contact</S.ContactTitle>);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should match the snapshot with dark theme', () => {
      const {container} = render(<S.ContactTitle $themeDark={true}>Contact</S.ContactTitle>);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should apply light theme color', () => {
      const {getByText} = render(<S.ContactTitle $themeDark={false}>Contact</S.ContactTitle>);
      const title = getByText('Contact');
      expect(title).toHaveStyle('color: #0a0a0a');
    });

    test('should apply dark theme color', () => {
      const {getByText} = render(<S.ContactTitle $themeDark={true}>Contact</S.ContactTitle>);
      const title = getByText('Contact');
      expect(title).toHaveStyle('color: #F0F2F1');
    });

    test('should have margin bottom', () => {
      const {getByText} = render(<S.ContactTitle $themeDark={false}>Contact</S.ContactTitle>);
      const title = getByText('Contact');
      expect(title).toHaveStyle('margin-bottom: 4px');
    });
  });

  describe('ContactText', () => {
    test('should match the snapshot with light theme', () => {
      const {container} = render(<S.ContactText $themeDark={false}>Email</S.ContactText>);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should match the snapshot with dark theme', () => {
      const {container} = render(<S.ContactText $themeDark={true}>Email</S.ContactText>);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should apply light theme color', () => {
      const {getByText} = render(<S.ContactText $themeDark={false}>Email</S.ContactText>);
      const text = getByText('Email');
      expect(text).toHaveStyle('color: #737373');
    });

    test('should apply dark theme color', () => {
      const {getByText} = render(<S.ContactText $themeDark={true}>Email</S.ContactText>);
      const text = getByText('Email');
      expect(text).toHaveStyle('color: #a3a3a3');
    });
  });

  describe('SocialLinks', () => {
    test('should match the snapshot', () => {
      const {container} = render(<S.SocialLinks />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should render as div element', () => {
      const {container} = render(<S.SocialLinks />);
      const element = container.firstChild as HTMLElement;
      expect(element.nodeName).toBe('DIV');
    });

    test('should have flex layout', () => {
      const {getByTestId} = render(<S.SocialLinks data-testid="social" />);
      const links = getByTestId('social');
      expect(links).toHaveStyle('display: flex');
      expect(links).toHaveStyle('align-items: center');
    });

    test('should have correct gap', () => {
      const {getByTestId} = render(<S.SocialLinks data-testid="social" />);
      const links = getByTestId('social');
      expect(links).toHaveStyle('gap: 24px');
    });
  });

  describe('SocialLink', () => {
    test('should match the snapshot', () => {
      const {container} = render(<S.SocialLink href="#" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should render as anchor element', () => {
      const {container} = render(<S.SocialLink href="#" />);
      const element = container.firstChild as HTMLElement;
      expect(element.nodeName).toBe('A');
    });

    test('should have flex layout centered', () => {
      const {getByTestId} = render(<S.SocialLink data-testid="link" href="#" />);
      const link = getByTestId('link');
      expect(link).toHaveStyle('display: flex');
      expect(link).toHaveStyle('align-items: center');
      expect(link).toHaveStyle('justify-content: center');
    });

    test('should have transition effect', () => {
      const {getByTestId} = render(<S.SocialLink data-testid="link" href="#" />);
      const link = getByTestId('link');
      expect(link).toHaveStyle('transition: transform 0.2s ease');
    });
  });

  describe('LogoIcon', () => {
    test('should match the snapshot', () => {
      const MockIcon = () => <svg />;
      const {container} = render(<S.LogoIcon icon={MockIcon} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should have correct icon dimensions', () => {
      const MockIcon = (props: any) => <svg data-testid="icon" {...props} />;
      const {getByTestId} = render(<S.LogoIcon icon={MockIcon} />);
      const icon = getByTestId('icon');
      expect(icon).toHaveAttribute('width', '32');
      expect(icon).toHaveAttribute('height', '32');
    });

    test('should have correct fill color', () => {
      const MockIcon = (props: any) => <svg data-testid="icon" {...props} />;
      const {getByTestId} = render(<S.LogoIcon icon={MockIcon} />);
      const icon = getByTestId('icon');
      expect(icon).toHaveAttribute('fill', '#000000');
    });
  });

  describe('HeartIcon', () => {
    test('should match the snapshot', () => {
      const {container} = render(<S.HeartIcon />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should have correct icon dimensions', () => {
      const {container} = render(<S.HeartIcon />);
      const icon = container.querySelector('svg');
      expect(icon).toHaveAttribute('width', '24');
      expect(icon).toHaveAttribute('height', '24');
    });

    test('should have correct color', () => {
      const {container} = render(<S.HeartIcon />);
      const icon = container.querySelector('svg');
      expect(icon).toHaveAttribute('fill', '#ef4444');
    });
  });

  describe('GithubIcon', () => {
    test('should match the snapshot with light theme', () => {
      const MockIcon = () => <svg />;
      const {container} = render(<S.GithubIcon icon={MockIcon} $themeDark={false} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should match the snapshot with dark theme', () => {
      const MockIcon = () => <svg />;
      const {container} = render(<S.GithubIcon icon={MockIcon} $themeDark={true} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should apply light theme icon color', () => {
      const MockIcon = (props: any) => <svg data-testid="icon" {...props} />;
      const {getByTestId} = render(<S.GithubIcon icon={MockIcon} $themeDark={false} />);
      const icon = getByTestId('icon');
      expect(icon).toHaveAttribute('fill', '#000000');
    });

    test('should apply dark theme icon color', () => {
      const MockIcon = (props: any) => <svg data-testid="icon" {...props} />;
      const {getByTestId} = render(<S.GithubIcon icon={MockIcon} $themeDark={true} />);
      const icon = getByTestId('icon');
      expect(icon).toHaveAttribute('fill', '#a3a3a3');
    });

    test('should have correct icon dimensions', () => {
      const MockIcon = (props: any) => <svg data-testid="icon" {...props} />;
      const {getByTestId} = render(<S.GithubIcon icon={MockIcon} $themeDark={false} />);
      const icon = getByTestId('icon');
      expect(icon).toHaveAttribute('width', '24');
      expect(icon).toHaveAttribute('height', '24');
    });
  });

  describe('LinkedinIcon', () => {
    test('should match the snapshot', () => {
      const MockIcon = () => <svg />;
      const {container} = render(<S.LinkedinIcon icon={MockIcon} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should have correct icon dimensions', () => {
      const MockIcon = (props: any) => <svg data-testid="icon" {...props} />;
      const {getByTestId} = render(<S.LinkedinIcon icon={MockIcon} />);
      const icon = getByTestId('icon');
      expect(icon).toHaveAttribute('width', '24');
      expect(icon).toHaveAttribute('height', '24');
    });
  });

  describe('EmailIcon', () => {
    test('should match the snapshot with light theme', () => {
      const MockIcon = () => <svg />;
      const {container} = render(<S.EmailIcon icon={MockIcon} $themeDark={false} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should match the snapshot with dark theme', () => {
      const MockIcon = () => <svg />;
      const {container} = render(<S.EmailIcon icon={MockIcon} $themeDark={true} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should apply light theme icon color', () => {
      const MockIcon = (props: any) => <svg data-testid="icon" {...props} />;
      const {getByTestId} = render(<S.EmailIcon icon={MockIcon} $themeDark={false} />);
      const icon = getByTestId('icon');
      expect(icon).toHaveAttribute('fill', '#000000');
    });

    test('should apply dark theme icon color', () => {
      const MockIcon = (props: any) => <svg data-testid="icon" {...props} />;
      const {getByTestId} = render(<S.EmailIcon icon={MockIcon} $themeDark={true} />);
      const icon = getByTestId('icon');
      expect(icon).toHaveAttribute('fill', '#a3a3a3');
    });

    test('should have correct icon dimensions', () => {
      const MockIcon = (props: any) => <svg data-testid="icon" {...props} />;
      const {getByTestId} = render(<S.EmailIcon icon={MockIcon} $themeDark={false} />);
      const icon = getByTestId('icon');
      expect(icon).toHaveAttribute('width', '32');
      expect(icon).toHaveAttribute('height', '32');
    });
  });

  describe('CopyrightContainer', () => {
    test('should match the snapshot', () => {
      const {container} = render(<S.CopyrightContainer />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should render as div element', () => {
      const {container} = render(<S.CopyrightContainer />);
      const element = container.firstChild as HTMLElement;
      expect(element.nodeName).toBe('DIV');
    });

    test('should have flex layout with space between', () => {
      const {getByTestId} = render(<S.CopyrightContainer data-testid="copyright" />);
      const container = getByTestId('copyright');
      expect(container).toHaveStyle('display: flex');
      expect(container).toHaveStyle('justify-content: space-between');
      expect(container).toHaveStyle('align-items: center');
    });

    test('should have correct padding', () => {
      const {getByTestId} = render(<S.CopyrightContainer data-testid="copyright" />);
      const container = getByTestId('copyright');
      expect(container).toHaveStyle('padding: 24px 0');
    });
  });

  describe('CopyrightText', () => {
    test('should match the snapshot with light theme', () => {
      const {container} = render(<S.CopyrightText $themeDark={false}>© 2025</S.CopyrightText>);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should match the snapshot with dark theme', () => {
      const {container} = render(<S.CopyrightText $themeDark={true}>© 2025</S.CopyrightText>);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should apply light theme color', () => {
      const {getByText} = render(<S.CopyrightText $themeDark={false}>© 2025</S.CopyrightText>);
      const text = getByText('© 2025');
      expect(text).toHaveStyle('color: #737373');
    });

    test('should apply dark theme color', () => {
      const {getByText} = render(<S.CopyrightText $themeDark={true}>© 2025</S.CopyrightText>);
      const text = getByText('© 2025');
      expect(text).toHaveStyle('color: #a3a3a3');
    });

    test('should have flex layout with gap', () => {
      const {getByText} = render(<S.CopyrightText $themeDark={false}>© 2025</S.CopyrightText>);
      const text = getByText('© 2025');
      expect(text).toHaveStyle('display: flex');
      expect(text).toHaveStyle('align-items: center');
      expect(text).toHaveStyle('gap: 4px');
    });
  });
});
