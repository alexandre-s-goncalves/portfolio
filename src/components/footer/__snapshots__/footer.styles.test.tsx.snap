// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer.styles BioText should match the snapshot with dark theme 1`] = `
.c0 {
  font-size: 16px;
  font-family: Inter-Regular;
  color: #a3a3a3;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
  line-height: 24px;
}

<p
  class="c0 "
>
  Bio text
</p>
`;

exports[`footer.styles BioText should match the snapshot with light theme 1`] = `
.c0 {
  font-size: 16px;
  font-family: Inter-Regular;
  color: #737373;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
  line-height: 24px;
}

<p
  class="c0 "
>
  Bio text
</p>
`;

exports[`footer.styles BrandTitle should match the snapshot with dark theme 1`] = `
.c0 {
  font-size: 18px;
  font-family: Inter-Regular;
  color: #F0F2F1;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
  line-height: 26px;
}

.c1 {
  display: flex;
  align-items: center;
  gap: 16px;
}

<h2
  class="c0 c1"
>
  Brand
</h2>
`;

exports[`footer.styles BrandTitle should match the snapshot with light theme 1`] = `
.c0 {
  font-size: 18px;
  font-family: Inter-Regular;
  color: #0a0a0a;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
  line-height: 26px;
}

.c1 {
  display: flex;
  align-items: center;
  gap: 16px;
}

<h2
  class="c0 c1"
>
  Brand
</h2>
`;

exports[`footer.styles ContactSection should match the snapshot 1`] = `
.c0 {
  flex: 1;
  display: flex;
  flex-direction: column;
  gap: 4px;
}

<div
  class="c0"
/>
`;

exports[`footer.styles ContactText should match the snapshot with dark theme 1`] = `
.c0 {
  font-size: 16px;
  font-family: Inter-Regular;
  color: #a3a3a3;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
  line-height: 24px;
}

<p
  class="c0 "
>
  Email
</p>
`;

exports[`footer.styles ContactText should match the snapshot with light theme 1`] = `
.c0 {
  font-size: 16px;
  font-family: Inter-Regular;
  color: #737373;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
  line-height: 24px;
}

<p
  class="c0 "
>
  Email
</p>
`;

exports[`footer.styles ContactTitle should match the snapshot with dark theme 1`] = `
.c0 {
  font-size: 16px;
  font-family: Inter-Regular;
  color: #F0F2F1;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
  line-height: 24px;
}

.c1 {
  margin-bottom: 4px;
}

<p
  class="c0 c1"
>
  Contact
</p>
`;

exports[`footer.styles ContactTitle should match the snapshot with light theme 1`] = `
.c0 {
  font-size: 16px;
  font-family: Inter-Regular;
  color: #0a0a0a;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
  line-height: 24px;
}

.c1 {
  margin-bottom: 4px;
}

<p
  class="c0 c1"
>
  Contact
</p>
`;

exports[`footer.styles ContentWrapper should match the snapshot 1`] = `
.c0 {
  display: flex;
  gap: 150px;
  width: 100%;
  padding-bottom: 32px;
  border-bottom: 1px solid #e5e5e5;
}

<div
  class="c0"
/>
`;

exports[`footer.styles CopyrightContainer should match the snapshot 1`] = `
.c0 {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 24px 0;
}

<div
  class="c0"
/>
`;

exports[`footer.styles CopyrightText should match the snapshot with dark theme 1`] = `
.c0 {
  font-size: 14px;
  font-family: Inter-Regular;
  color: #a3a3a3;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
  line-height: 20px;
}

.c1 {
  display: flex;
  align-items: center;
  gap: 4px;
}

<p
  class="c0 c1"
>
  © 2025
</p>
`;

exports[`footer.styles CopyrightText should match the snapshot with light theme 1`] = `
.c0 {
  font-size: 14px;
  font-family: Inter-Regular;
  color: #737373;
  margin-top: 0px;
  font-weight: normal;
  text-transform: none;
  cursor: text;
  line-height: 20px;
}

.c1 {
  display: flex;
  align-items: center;
  gap: 4px;
}

<p
  class="c0 c1"
>
  © 2025
</p>
`;

exports[`footer.styles EmailIcon should match the snapshot with dark theme 1`] = `<svg />`;

exports[`footer.styles EmailIcon should match the snapshot with light theme 1`] = `<svg />`;

exports[`footer.styles FooterContainer should match the snapshot 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  width: 100%;
  background-color: #ffffff;
  padding: 32px 120px;
  border-top: 1px solid #e5e5e5;
  box-sizing: border-box;
}

<footer
  class="c0"
/>
`;

exports[`footer.styles GithubIcon should match the snapshot with dark theme 1`] = `<svg />`;

exports[`footer.styles GithubIcon should match the snapshot with light theme 1`] = `<svg />`;

exports[`footer.styles HeartIcon should match the snapshot 1`] = `
<svg
  class=""
  fill="#ef4444"
  height="24"
  style="min-width: 24px; min-height: 24px; max-width: 24px; max-height: 24px;"
  width="24"
/>
`;

exports[`footer.styles InfoSection should match the snapshot 1`] = `
.c0 {
  flex: 1;
  display: flex;
  flex-direction: column;
  max-width: 500px;
  gap: 16px;
}

<div
  class="c0"
/>
`;

exports[`footer.styles LinkedinIcon should match the snapshot 1`] = `<svg />`;

exports[`footer.styles LogoIcon should match the snapshot 1`] = `<svg />`;

exports[`footer.styles SocialLink should match the snapshot 1`] = `
.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  transition: transform 0.2s ease;
}

.c0:hover {
  transform: scale(1.1);
}

<a
  class="c0"
  href="#"
/>
`;

exports[`footer.styles SocialLinks should match the snapshot 1`] = `
.c0 {
  display: flex;
  align-items: center;
  gap: 24px;
}

<div
  class="c0"
/>
`;
